core/opendaq/device/include/opendaq/connection_status_container_impl.h

```diff
@@ -0,0 +1,336 @@
+    SizeT streamingConnectionsCounter;
+};
+
+inline ConnectionStatusContainerImpl::ConnectionStatusContainerImpl()
+    : ConnectionStatusContainerImpl(nullptr, nullptr)
+{
+}
+
+inline ConnectionStatusContainerImpl::ConnectionStatusContainerImpl(const ContextPtr& context, const ProcedurePtr& coreEventCallback)
+    : Super(coreEventCallback)
+    , context(context)
+    , statusNameAliases(Dict<IString, IString>())
+    , streamingConnectionsCounter(0)
+{
+}
+
+inline ErrCode ConnectionStatusContainerImpl::getStatus(IString* name, IEnumeration** value)
+{
+    OPENDAQ_PARAM_NOT_NULL(name);
+    OPENDAQ_PARAM_NOT_NULL(value);
+
+    const auto nameObj = StringPtr::Borrow(name);
+    if (nameObj == "")
+        return OPENDAQ_ERR_INVALIDPARAMETER;
+
+    std::scoped_lock lock(sync);
+
+    for (const auto& [connectionString, nameAlias] : statusNameAliases)
+    {
+        if (nameAlias == nameObj && statuses.hasKey(connectionString))
+        {
+            *value = statuses.get(connectionString).addRefAndReturn();
+            return OPENDAQ_SUCCESS;
+        }
+    }
+
+    return OPENDAQ_ERR_NOTFOUND;
+}
+
+inline ErrCode ConnectionStatusContainerImpl::getStatuses(IDict** statuses)
+{
+    OPENDAQ_PARAM_NOT_NULL(statuses);
+
+    auto statusesAliased = Dict<IString, IEnumeration>();
+
+    std::scoped_lock lock(sync);
+
+    for (const auto& [connectionString, nameAlias] : statusNameAliases)
+    {
+        if (this->statuses.hasKey(connectionString))
+            statusesAliased[nameAlias] = this->statuses.get(connectionString);
+    }
+
+    *statuses = statusesAliased.detach();
+    return OPENDAQ_SUCCESS;
+}
+
+inline ErrCode ConnectionStatusContainerImpl::addConfigurationConnectionStatus(IString* connectionString, IEnumeration* initialValue)
+{
+    OPENDAQ_PARAM_NOT_NULL(connectionString);
+    OPENDAQ_PARAM_NOT_NULL(initialValue);
+
+    const auto connectionStringObj = StringPtr::Borrow(connectionString);
+    if (connectionStringObj == "")
+        return OPENDAQ_ERR_INVALIDPARAMETER;
+
+    std::scoped_lock lock(sync);
+
+    if (statusNameAliases.hasKey(ConfigurationConnectionStatusAlias) ||
+        statuses.hasKey(connectionStringObj))
+        return OPENDAQ_ERR_ALREADYEXISTS;
+
+    statuses[connectionStringObj] = initialValue;
+    statusNameAliases[connectionStringObj] = ConfigurationConnectionStatusAlias;
+
+    return OPENDAQ_SUCCESS;
+}
+
+inline ErrCode ConnectionStatusContainerImpl::addStreamingConnectionStatus(IString* connectionString, IEnumeration* initialValue, IBaseObject* streamingObject)
+{
+    OPENDAQ_PARAM_NOT_NULL(connectionString);
+    OPENDAQ_PARAM_NOT_NULL(initialValue);
+
+    const auto connectionStringObj = StringPtr::Borrow(connectionString);
+    if (connectionStringObj == "")
+        return OPENDAQ_ERR_INVALIDPARAMETER;
+
+    std::scoped_lock lock(sync);
+
+    if (statuses.hasKey(connectionStringObj))
+        return OPENDAQ_ERR_ALREADYEXISTS;
+
+    ++streamingConnectionsCounter;
```

Comment:
Counter is incremented in addStreamingConnectionStatus() but not decremented in removeStreamingConnectionStatus() is this intentional?

Reply:
Yes, the value is used to create numeration suffix in streaming status alias and it is not decremented intentionally to make sure any new alias has unique number.